src/lib.rs

```diff
@@ -197,7 +197,7 @@ impl Progress {
                 let mut b = &mut self.bars[bar.0];
                 let w = (term_width / 2) - 7;
                 let (data, unit) = denomination(b.curr);
-                let diff = 100 * (b.curr - b.prev) / b.total;
+                let diff = 100 * ((b.curr - b.prev) / b.total);
```

Comment:
Same issue here, the experession in parentheses evaluates to 1 if `(b.curr - b.prev)  == b.total` and 0 otherwise. Therefore, the progressbar stays at 0 and then suddenly jumps to 100%.

Reply:
I'll fix this tomorrow.